web/src/reducers/auth.js

```diff
@@ -0,0 +1,29 @@
+import {
+  LOGIN_REQUEST,
+  LOGIN_SUCCESS,
+  LOGIN_FAILURE,
+  LOGOUT_SUCCESS
+} from '../actions/auth';
+
+const initialState = {
+  fetching: false,
+  authenticated: false,
+  error: ''
+};
+
+const auth = (state = initialState, action) => {
+  switch (action.type) {
+    case LOGIN_REQUEST:
+      return { ...state, fetching: true, authenticated: false, error: '' };
```

Comment:
I _*love*_ this pattern.  I used to use [updeep](https://npm.im/updeep) to do deep cloning and it was kind of a hassle.  This is so much nicer.

Reply:
👍 🎉